ipykernel/ipkernel.py

```diff
@@ -469,7 +469,8 @@ def do_inspect(self, code, cursor_pos, detail_level=0):
             reply_content['data'].update(
                 self.shell.object_inspect_mime(
                     name,
-                    detail_level=detail_level
+                    detail_level=detail_level,
+                    omit_sections=omit_sections,
```

Comment:
If we guard this with a `TypeError` catch and then retry without `omit_sections` then we maintain backwards compatibility.

Reply:
You can also be explicit and check against `IPython.version_info` tuple, this will make it obvious which codepath to cleanup when we stop supporting IPython 7.x. If we use try/except we may catch too many error and it will not be obvious in 2/3 years what's happening here.

Reply:
@Carreau and @blink1073 thanks for the tips!!! I have prepared a commit to look at `version_info`, and will push it shortly. A small question: if we do it that way, ipykernel master to not use this until IPython v8 is released—are we ok with that? (Meanwhile the `TypeError` approach would let both modules interoperate in master.)

Reply:
Pushed 0bdfd0bf9ae8e1dd61802ac90f16a9d8d4ed74af, very much open to your feedback 🙇🙏!

Reply:
The version number of master is already 8.0.0.dev,  so it will (should) take this branch, and I'm likely to release 8.0 end of december/early 2022. The only case where this would fail is if someone is using and older commit of IPython master branch, and ipykernel after this is merged. 

In that case I believe the person is likely a developer and would now how to fix it/update their repositories.

Reply:
Ahh of course, makes total sense! Sorry for being slow 😅, I realize now I was looking at the Conda-installed copy of IPython's `release.py`, not master. I pushed this improvement, thanks so much for your patience and very kind assistance 🙏!

Reply:
No problem, we don't expect you to be aware of all the informations and when what is released and what is our future plan.